tools/ci_import

```
@@ -0,0 +1,94 @@
+        folder = f"/tmp/{job.get_id()}/"
+        os.makedirs(folder, exist_ok=True)
+        filename = os.path.join(folder, "artifact.zip")
+        if verbose:
+            print(f"    downloading artifact.zip to {filename} for job {job.get_id()}")
+        try:
+            with open(filename, "wb") as f:
+                project.jobs.get(job.get_id()).artifacts(streamed=True, action=f.write)
+            z = zipfile.ZipFile(filename)
+            z.extractall(folder)
+            if verbose:
+                print(f"    image information imported for {job.get_id()}")
+            try:
+                imis = os.listdir(os.path.join(folder, IMI_FOLDER))
+            except FileNotFoundError:
+                continue
+            # for each downloaded image info, update the test cases
+            for imi in imis:
+                imi_path = os.path.join(folder, IMI_FOLDER, imi)
+                if verbose:
+                    print(f"    {imi}:")
+                # load image info
+                with open(imi_path, "r", encoding="utf-8") as f:
+                    imi_data = json.load(f)
+                # load test case
+                test_case_path = os.path.join("manifest-db", imi)
+                with open(test_case_path, "r", encoding="utf-8") as f:
+                    tc_data = json.load(f)
+                # update test case
+                tc_data["image-info"] = imi_data
+                try:
+                    if verbose:
+                        print(f"        updating {test_case_path}:", end=" ")
+                    with open(test_case_path, "w", encoding="utf-8") as f:
+                        json.dump(tc_data, f, indent=2)
+                    if verbose:
+                        print("✅")
+                except Exception as e:  # pylint: disable=[broad-except]
+                    if verbose:
+                        print(f"{e} ❌")
+        except gitlab.exceptions.GitlabGetError as e:
+            if verbose:
```

Same concern about an ignored exception here.

---

Sometimes, for some jobs there's no artifacts to download, therefore we get a 404. The exception outputs some stuff anyway. At least in verbose mode. The tool should not crash here.